### DISCUSSION TITLE
How to get fields of children when querying the parent? :(

### DISCUSSION THREAD

**Discussion by ebsaral** [nocodb/nocodb#696] [Q&A] [answered]
Hello! We recently started using NocoDB company-wide, and are very excited to build our APIs on top of NocoDB yet we are struggling to really understand how to use the REST API. 

I would like to demonstrate a simple DB structure and ask for your guidance to create the correct query: 

table: label_objects
fields:
- title: str
- color: str
- id: PK

table: labels
fields: 
- id: PK
- label_object_id: FK to label_objects defined above

When I run `https://{domain}/api/v1/labels`, I get this result:

```json
[
  {
    "id": 1,
    "label_object_id": 1,
    "label_objectsRead": {
      "id": 1
    }
  }
]
```

How can I achieve the result below by getting `title` and `color` fields of the parent (label_objects) in the result?

```json
[
  {
    "id": 1,
    "label_object_id": 1,
    "label_objectsRead": {
      "id": 1,
      "title": "Test title",
      "color": "#000000"
    }
  }
]
```

Could you please share an example query to get this result? 

It also applies to getting children' fields. Same issue is there.

The documentation is not clear how to do this and trying different scenarios in Swagger ends up with an error:

```json
{
  "msg": "Cannot read property 'tnPath' of undefined"
}
```

If you could also show an example of how to hide/show certain parents/children tables in an example query, that would be great as well. So far, I also couldn't figure it out from the docs.

This issue is a blocker for us benefiting from NocoDB with full capacity. :( 

Thank you for your help and advices!!!

**Comment by pranavxc** [comment, marked as answer]
By default, we are extracting all the relations and for the nested record we are only getting **primary key** and **primary value** by default.

<hr>

Now in order to get more nested fields, you can pass fields query parameter but it's slightly different, you need to pass it as follows:

#### Key will be in the following format - 

```
 {first_letter_of_relation_type}fields{index}
```

OR

```
 {first_letter_of_relation_type}f{index_of}
```
| Variable| Value |
|---|---|
| **first_letter_of_relation_type** | In case of belongs to it will be `b`, for has many `h` and for many to many `m` |
|  **index** | index of relation within relation type, index starts from `1` |

 



> Now the value of the query parameter will be comma-separated column names


#### Example 1: 


With your own example if you want to extract `color` and `title` from first belongs to relation you need to pass the following query parameter.

```
bfields1=title,color
```

or

```
bf1=title,color
```

<hr>

Now if you have multiple relations and you want only one of the relations among the 2 then you can explicitly specify query parameters to restrict extracting all nested data.  

Available query params keys are 

| Key | Value | Example | Default behavior | 
| -- | -- | -- | -- |
| bt | comma-separated belongs to tables | `table1,table2` | All the belongs to table | 
| mm | comma-separated many to many to tables |  `table1,table2` | All the many to many to table |  
| hm | comma-separated has many to tables |   `table1,table2` | All the has many to table | 

#### Example 2:

If you have 2 belongs to table(`b` and `c`) and you just want data from `c` and from `c` you want `title` and `body` then you can pass the following query params.


```
bt=c&bfields1=title,body
```
